### DISCUSSION TITLE
fastapi server and process number.

### DISCUSSION THREAD

**Discussion by JuHyung-Son** [fastapi/fastapi#6539] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@router.get('/{id}',
             status_code=fastapi.status.HTTP_200_OK)
def get(id: str):
    time.sleep(100)
    return Response()
```

### Description

With this code, I ran server with 1 worker.
` uvicorn app:app --reload --workers 1 --host 0.0.0.0 --port 8000 --log-config log_config.yaml --log-level debug`

I expect only one request is available for one worker. So if I request `/{id}` above. no other requests are possible for 100 seconds.

However, it is possible to send another requests while time.sleep(100).
how is it possible?

### Operating System

macOS

### FastAPI Version

0.65.2

### Python Version

3.8

**Comment by TheoBabilon** [comment, marked as answer]
Hi JuHyung-Son,

Any path operation function defined as "def" (synchronous function) will be automatically run in an external threadpool to make it non-blocking:
"**When you declare a path operation function with normal def instead of async def, it is run in an external threadpool that is then awaited, instead of being called directly (as it would block the server).**", quoted from https://fastapi.tiangolo.com/async/

I'm guessing this is why you are still able to make other requests.
You might want to try defining it as "**async def**" (even though it makes no sense here because you're not awaiting anything in the function) to try and confirm the behavior - that should become blocking as you're not using an async call to asyncio's sleep function (await asyncio.sleep(x)).
